ecmascript/transforms/typescript/src/strip.rs

```diff
@@ -2217,6 +2217,34 @@ where
         d.visit_mut_children_with(self);
         d.definite = false;
     }
+
+    fn visit_mut_pat_or_expr(&mut self, node: &mut PatOrExpr) {
+        // Coerce bindingident to assign expr where parenthesis exists due to TsAsExpr
+        // like (warn as any) = v;
+        // We want to do this _before_ visiting children, otherwise handle_expr
+        // wipes out TsAsExpr and there's no way to distinguish between
+        // plain (x) = y vs. (x as any) = y;
+        match node {
+            PatOrExpr::Pat(pat) => match &**pat {
```

Comment:
I'm feeling this nested match to deref Box is bit unidomatic, potentially due to lacks of my knowledge and would like to simplify if possible.

Reply:
It's possible with nightly rustc, but there are many swc users who use stable rustc, so it can't be simplified for now.

Reply:
Yes, I used box signatuer for the internal plugin experiment. Sad it's not stable.